bears/c_languages/GNUIndentBear.py

```diff
@@ -23,18 +23,160 @@ class GNUIndentBear:
+            Forces variables names to be aligned in column ``n`` with
+            ``n = declaration_indent``  in declaration.
+
+            Example:
+                If ``8``,
+                ```
+                int a;
+                float b;
+                ```
+                changes to
+                ```
+                int     a;
+                float   b;
+                ```
+        :param brace_indent:
+            Specifies the number of spaces by which braces are indented. Its
+            default value is 2.
+        :param gnu_style:
+            Uses GNU coding style.
+        :param k_and_r_style:
+            Uses Kernighan & Ritchie coding style.
+        :param linux_style:
+            Uses Linux coding style.
+        :param tab_width:
+            Number of spaces per indent level.
+        :param indent_cli_options:
+            Any command line options the indent binary understands. They
+            will be simply passed through.
         """
-        args = ("--no-tabs" if use_spaces else "--use-tabs",
-                "--line-length", str(max_line_length),
-                "--indent-level", str(tab_width),
-                "--tab-size", str(tab_width))
-        return args + tuple(shlex.split(indent_cli_options))
+        indent_options = ("--no-tabs" if use_spaces else "--use-tabs",
+                          "--line-length", str(max_line_length),
+                          "--indent-level", str(tab_width),
+                          "--tab-size", str(tab_width), )
+        indent_options += (("--cuddle-do-while",)
+                           if while_and_brace_on_same_line
+                           else ("--dont-cuddle-do-while",))
+        indent_options += (("--swallow-optional-blank-lines",)
+                           if delete_optional_blank_lines else ("-nsob",))
+        indent_options += (("--blank-lines-after-declarations",)
+                           if blank_lines_after_declarations else ("-nbad",))
+        indent_options += (("--blank-lines-after-commas",)
+                           if blank_lines_after_commas else ("-nbc",))
+        indent_options += (("--blank-lines-after-procedures",)
+                           if blank_lines_after_procedures else ("-nbap",))
+        indent_options += (("-di" + str(declaration_indent),)
+                           if declaration_indent != 0 else ())
+        indent_options += (("--case-indentation"+str(case_indentation),)
+                           if case_indentation != 0 else ())
+        indent_options += (("--space-special-semicolon",)
+                           if space_before_semicolon_after_empty_loop
+                           else ("-nss",))
+        indent_options += ("--brace-indent"+str(brace_indent),)
+        indent_options += ("--gnu-style",) if gnu_style else ()
+        indent_options += ("--k-and-r-style",) if k_and_r_style else ()
+        indent_options += ("--linux-style",) if linux_style else ()
```

Comment:
if a option is provided explicilty, i will override the one provided by the
style

2016-06-16 20:47 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

> In bears/c_languages/GNUIndentBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/519#discussion_r67411819>
> :
>
> > +                           if blank_lines_after_declarations else ("-nbad",))
> > +        indent_options += (("--blank-lines-after-commas",)
> > +                           if blank_lines_after_commas else ("-nbc",))
> > +        indent_options += (("--blank-lines-after-procedures",)
> > +                           if blank_lines_after_procedures else ("-nbap",))
> > +        indent_options += (("-di" + str(declaration_indent),)
> > +                           if declaration_indent != 0 else ())
> > +        indent_options += (("--case-indentation"+str(case_indentation),)
> > +                           if case_indentation != 0 else ())
> > +        indent_options += (("--space-special-semicolon",)
> > +                           if space_before_semicolon_after_empty_loop
> > +                           else ("-nss",))
> > +        indent_options += ("--brace-indent"+str(brace_indent),)
> > +        indent_options += ("--gnu-style",) if gnu_style else ()
> > +        indent_options += ("--k-and-r-style",) if k_and_r_style else ()
> > +        indent_options += ("--linux-style",) if linux_style else ()
>
> hm, those style arguments are a combination of the others, I wonder if the
> defaults we have above override those potentially and thus do not result in
> the wanted style?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/519/files/d88abd9e6371173007240bef6fb8a02fa123726a#r67411819>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xotj5uTX1VIQCkUAgQn6R53kO5yVyks5qMag7gaJpZM4I02l0>
> .
>